include/swift/AST/AutoDiff.h

```diff
@@ -262,11 +269,19 @@ using SILDifferentiabilityWitnessKey = std::pair<StringRef, AutoDiffConfig>;
 /// Automatic differentiation utility namespace.
 namespace autodiff {
 
-/// Appends the subset's parameter's types to `results`, in the order in
-/// which they appear in the function type.
-void getSubsetParameterTypes(IndexSubset *indices, AnyFunctionType *type,
-                             SmallVectorImpl<Type> &results,
-                             bool reverseCurryLevels = false);
+/// Given a function type, returns its semantic result type: either the formal
+/// result type or an `inout` parameter.
+///
+/// The function type may have at most two parameter lists.
+///
+/// Sets `hasMultipleOriginalSemanticResults` to true iff there are multiple
+/// semantic results.
+///
+/// Remaps the original semantic result using `genericEnv`, if specified.
+AutoDiffSemanticFunctionResultType
+getFunctionSemanticResultType(AnyFunctionType *functionType,
+                              bool &hasMultipleSemanticResults,
```

Comment:
This argument is not being used by any caller, and AD isn't going to support multiple semantic results any time soon.

Reply:
Nvm, I'm seeing a user now. But this function still seems problematic in that when there are multiple semantic results it is unclear which result the returned value is.  Why not, instead, define this function as:

```swift
void getFunctionSemanticResultTypes(
    AnyFunctionType *functionType,
    GenericEnvironment *genericEnv = nullptr,
    SmallVectorImpl<AutoDiffSemanticFunctionResultType> &result)
```

Reply:
`getFunctionSemanticResultTypes` is indeed nicer than `getFunctionSemanticResultType` with the `hasMultipleSemanticResults` boolean.

Done:
```cpp
/// Given a function type, collects its semantic result types in type order
/// into `result`: first, the formal result type (if non-`Void`), followed by
/// `inout` parameter types.
///
/// The function type may have at most two parameter lists.
///
/// Remaps the original semantic result using `genericEnv`, if specified.
void getFunctionSemanticResultTypes(
    AnyFunctionType *functionType,
    SmallVectorImpl<AutoDiffSemanticFunctionResultType> &result,
    GenericEnvironment *genericEnv = nullptr);
```